Userland/Libraries/LibC/unistd.cpp

```diff
@@ -419,9 +419,9 @@ ssize_t pwrite(int fd, void const* buf, size_t count, off_t offset)
 }
 
 // Note: Be sure to send to directory_name parameter a directory name ended with trailing slash.
-static int ttyname_r_for_directory(char const* directory_name, dev_t device_mode, ino_t inode_number, char* buffer, size_t size)
+static int ttyname_r_for_directory(StringView directory_name, dev_t device_mode, ino_t inode_number, char* buffer, size_t size)
 {
-    DIR* dirstream = opendir(directory_name);
+    DIR* dirstream = opendir(directory_name.characters_without_null_termination());
```

Comment:
This doesn't look right. `opendir()` expects a null-terminated string, yet we're calling it with something we get from `characters_without_null_termination()` :sweat_smile:

Reply:
So we have to use raw pointer, because neither `StringView` nor `String` has accessor for null terminated cstring?